client/client-multi/client-c/src/platform-impl/kaa_tcp_channel.c

```diff
@@ -844,19 +814,77 @@ kaa_error_t kaa_tcp_channel_check_keepalive(kaa_transport_channel_interface_t *s
+                break;
+            case RET_STATE_VALUE_READY:
+                tcp_channel->access_point.state = AP_RESOLVED;
+                KAA_LOG_TRACE(tcp_channel->logger, KAA_ERR_NONE, "Kaa TCP channel new access point [0x%08X] destination resolved"
+                                                                                                    , tcp_channel->access_point.id);
+                error_code = kaa_tcp_channel_connect_access_point(tcp_channel);
+                if (error_code) {
+                    KAA_LOG_ERROR(tcp_channel->logger, error_code, "Kaa TCP channel new access point [0x%08X] failed to connect"
+                                                                                                            , tcp_channel->access_point.id);
+                    if (tcp_channel->event_callback)
+                        tcp_channel->event_callback(tcp_channel->event_context
+                                              , SOCKET_CONNECTION_ERROR
+                                              , tcp_channel->access_point.socket_descriptor);
+
+                    error_code = kaa_tcp_channel_on_access_point_failed(tcp_channel);
+                }
+                break;
+            case RET_STATE_VALUE_ERROR:
+                tcp_channel->access_point.state = AP_NOT_SET;
+                error_code = KAA_ERR_TCPCHANNEL_AP_RESOLVE_FAILED;
+                KAA_LOG_ERROR(tcp_channel->logger, error_code, "Kaa TCP channel new access point [0x%08X] hostname resolve failed"
+                                                                                                    , tcp_channel->access_point.id);
+                error_code = kaa_tcp_channel_on_access_point_failed(tcp_channel);
+                break;
+            case RET_STATE_BUFFER_NOT_ENOUGH:
+                tcp_channel->access_point.state = AP_NOT_SET;
+                error_code = KAA_ERR_TCPCHANNEL_AP_RESOLVE_FAILED;
+                KAA_LOG_ERROR(tcp_channel->logger, error_code, "Kaa TCP channel new access point [0x%08X] hostname resolve failed. "
+                                                                        "Address buffer is not enough", tcp_channel->access_point.id);
+                break;
+        }
+    } else {
+
+        KAA_LOG_INFO(tcp_channel->logger, KAA_ERR_NONE, "Kaa TCP channel [0x%08X] checking keepalive"
+                                                                        , tcp_channel->access_point.id);
 
-    if (tcp_channel->keepalive.keepalive_interval == 0
-            || tcp_channel->channel_state != KAA_TCP_CHANNEL_AUTHORIZED) {
-        return error_code;
-    }
+        if (tcp_channel->keepalive.keepalive_interval == 0
+                || tcp_channel->channel_state != KAA_TCP_CHANNEL_AUTHORIZED) {
+            return error_code;
+        }
+
+        kaa_time_t interval = (kaa_time_t) ext_get_systime() - (kaa_time_t) tcp_channel->keepalive.last_sent_keepalive;
```

Comment:
Please, use the KAA_TIME() macros from "platform/time.h" and the corresponding typedef.